reactor-core/src/main/java/reactor/core/publisher/Traces.java

```diff
@@ -52,9 +52,9 @@
 
 	static {
 		String[] strategyClasses = {
-				Traces.class.getName() + "$StackWalkerCallSiteSupplierFactory",
-				Traces.class.getName() + "$SharedSecretsCallSiteSupplierFactory",
-				Traces.class.getName() + "$ExceptionCallSiteSupplierFactory",
+			StackWalkerCallSiteSupplierFactory.class.getName(),
```

Comment:
Not sure it will be working that way since `.class` should likely load a class before grabbing its name. Which says we may have an exception there as well as the code starting from line 61 starts making zero sense as the class is loaded beforehand

Reply:
This way it will indeed trigger the classloading. It _may_ work, but, if you ever add e.g. JDK9+ class to the signature of these classes, the classloading will fail. Although, testing FTW, just make sure that you have it covered (by testing on both JDK8 and JDK9+)

Reply:
`SomeClass.class` and `SomeClass.class.getName()` do not load the class, it appears.
And the static block is only ran at class loading, so eg. the above won't attempt to load `StackWalker`.

When the stream code does invoke the _constructor_ reflectively, it loads the class, tries to load `StackWalker` on which it depends, and throws the `NoClassDefFoundError`.

Reply:
see https://docs.oracle.com/javase/specs/jls/se9/html/jls-12.html#jls-12.4.1

Reply:
we can actually even directly put the `Class`es in the array, I think. `Class.forName("className")` _does_ however initializes the class (but you can also use the overload `Class.forName(name, false, this.getClass().getClassLoader())` to avoid that).